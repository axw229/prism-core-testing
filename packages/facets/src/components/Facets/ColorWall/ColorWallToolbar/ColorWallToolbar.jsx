/* eslint-disable */
// @flow
/* eslint-disable react/jsx-no-bind */
import React, { type Element, useContext, useEffect, useRef } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useRouteMatch, NavLink, useHistory, Link } from 'react-router-dom'
import { filterByFamily, filterBySection } from 'src/store/actions/loadColors'
import { FormattedMessage, useIntl } from 'react-intl'
import { Menu, MenuItem, Wrapper, Button } from 'react-aria-menubutton'
import at from 'lodash/at'
import kebabCase from 'lodash/kebabCase'
import difference from 'lodash/difference'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { AutoSizer } from 'react-virtualized'
import { MODE_CLASS_NAMES } from '../shared'
import ButtonBar from 'src/components/GeneralButtons/ButtonBar/ButtonBar'
import { generateColorWallPageUrl } from 'src/shared/helpers/ColorUtils'
import ColorWallContext from '../ColorWallContext'
import ConfigurationContext, {
  type ConfigurationContextType
} from 'src/contexts/ConfigurationContext/ConfigurationContext'
import { navigateToIntendedDestination, setIsColorWallModallyPresented } from 'src/store/actions/navigation'
import './ColorWallMenuBar.scss'
import '../../../GeneralButtons/ButtonBar/ButtonBar.scss'
import omitPrefix from 'src/shared/utils/omitPrefix.util'
import * as GA from 'src/analytics/GoogleAnalytics'
import { GA_TRACKER_NAME_BRAND } from 'src/constants/globals'
import useGroupsAndSubgroups from 'src/shared/hooks/useGroupsAndSubgroups'

const PATH_END_FAMILY = 'family/'
const menuBarPrefix = 'menu-bar'
const menuBarItemList = `${menuBarPrefix}__menu-item`
const menuBarActiveList = `${menuBarItemList}--active`
const menuBarInactive = `${menuBarItemList}--inactive`

type SelectPropsT = {
  activeFamily?: string,
  activeSection?: string,
  disabled?: boolean,
  options: { label: string, link: string }[],
  onSelectOpened?: (boolean) => void,
  placeholderText: string
}

const Select = ({
  activeFamily,
  activeSection,
  placeholderText,
  options,
  disabled = false,
  onSelectOpened
}: SelectPropsT) => (
  <Wrapper
    className={`${MODE_CLASS_NAMES.CELL} ${MODE_CLASS_NAMES.RIGHT} ${menuBarPrefix}`}
    onMenuToggle={({ isOpen }) => onSelectOpened?.(isOpen)}
  >
    <Button className={`${menuBarPrefix}__button`} disabled={disabled}>
      <span className={`${menuBarPrefix}__button-copy`}>{omitPrefix(placeholderText)}</span>
      <FontAwesomeIcon className='close-icon-svg' icon={['fa', 'angle-down']} pull='right' />
    </Button>
    <Menu className={`${menuBarPrefix}__menu`}>
      {options.map(({ label, link }) => (
        <MenuItem className={`${menuBarItemList}`} key={label} text={omitPrefix(label)} value={label}>
          <Link
            className={`${menuBarPrefix}__menu-link ${
              ((activeSection ?? activeFamily) || 'All') === label ? menuBarActiveList : menuBarInactive
            }`}
            to={{ data: activeSection ?? label, pathname: link, state: label }}
          >
            <span className={MODE_CLASS_NAMES.DESC}>{omitPrefix(label)}</span>
          </Link>
        </MenuItem>
      ))}
    </Menu>
  </Wrapper>
)

type ColorFamilyMenuBtnsT = {
  showAll?: boolean,
  section: string,
  families: string[]
}

const ColorFamilyMenuBtns = ({ showAll = false, section, families = [] }: ColorFamilyMenuBtnsT) => {
  const { brandId }: ConfigurationContextType = useContext(ConfigurationContext)

  if (families.length) {
    return (
      <>
        {showAll ? (
          <ButtonBar.Button
            isActive={(match, location) => {
              if (!match) {
                return false
              }

              return !!location.pathname.match(new RegExp(`${match.url}/?(/color/.*)?/?$`))
            }}
            style={{ justifyContent: 'center', width: '100%' }}
            to={generateColorWallPageUrl(section)}
          >
            <span className={MODE_CLASS_NAMES.DESC}>All</span>
          </ButtonBar.Button>
        ) : null}
        {families.map((name) => (
          <ButtonBar.Button
            key={name}
            onClick={() =>
              GA.event(
                { category: 'Color Wall', action: 'Color Family Menu Click', label: name },
                GA_TRACKER_NAME_BRAND[brandId]
              )
            }
            style={{ justifyContent: 'center', width: '100%' }}
            to={generateColorWallPageUrl(section, name)}
          >
            <span className={MODE_CLASS_NAMES.DESC}>{omitPrefix(name)}</span>
          </ButtonBar.Button>
        ))}
      </>
    )
  }

  return null
}

const ColorWallToolbar = () => {
  const { messages = {} } = useIntl()
  const { hiddenSections } = useContext(ColorWallContext)
  const {
    groups: sections,
    group: activeSection,
    subgroups: families,
    subgroup: activeFamily
  } = useGroupsAndSubgroups(hiddenSections)
  const {
    alwaysShowColorFamilies,
    colorWall = {},
    cvw = {},
    uiStyle
  } = useContext<ConfigurationContextType>(ConfigurationContext)
  const { closeBtn = {} } = cvw
  const { showArrow: closeBtnShowArrow = true, text: closeBtnText = <FormattedMessage id='CLOSE' /> } = closeBtn
  const {
    path,
    params: { section, family }
  } = useRouteMatch()
  const { primeColorWall } = useSelector((state) => state.colors)
  const dispatch = useDispatch()
  const history = useHistory()
  const isFamilyView: boolean = !!family || path.endsWith(PATH_END_FAMILY)
  const visibleSections: string[] = sections

  // This should have been set by staging action...
  const shouldShowCloseButton = useSelector((store) => store.isColorwallModallyPresented)

  // keeps redux store and url in sync for family and section data
  useEffect(() => {
    if (activeSection === primeColorWall && section && section !== kebabCase(primeColorWall)) {
      dispatch(filterBySection(section))
    }
    if (!activeFamily && family) {
      dispatch(filterByFamily(family))
    }
  }, [])

  const searchColorBtn: Element<any> = (
    <ButtonBar.Button to={`${generateColorWallPageUrl(section, family)}search/`}>
      <span className={MODE_CLASS_NAMES.DESC}>
        {colorWall.searchColor ?? <FormattedMessage id='SEARCH.SEARCH_COLOR' />}
      </span>
      <FontAwesomeIcon className='color-families-svg' icon={['fa', 'search']} pull='left' />
    </ButtonBar.Button>
  )

  const colorFamilyMenu = useRef(null)

  if (uiStyle === 'minimal') {
    return (
      <div className={MODE_CLASS_NAMES.BASE}>
        <div className={`${MODE_CLASS_NAMES.COL} ${menuBarPrefix}-minimal`}>
          <button
            className='search'
            onClick={() => history.push(`${generateColorWallPageUrl(section, family)}search/`)}
          >
            <FontAwesomeIcon icon={['fal', 'search']} size='lg' />
            <span style={{ opacity: '0.5', marginLeft: '15px' }}>
              <FormattedMessage id='SEARCH.SEARCH_FOR_A_COLOR' />
            </span>
          </button>
          <Select
            disabled={families.length < 2}
            placeholderText={activeFamily || messages['EXPLORE_COLOR_FAMILIES']}
            options={families
              .filter((f) => f !== activeFamily)
              .map((label) => ({ label, link: generateColorWallPageUrl(section, label) }))}
          />
          <Select
            disabled={visibleSections.length < 2}
            placeholderText={
              activeSection === primeColorWall || !visibleSections.includes(activeSection)
                ? colorWall.selectSectionText ?? messages['EXPLORE_COLLECTIONS']
                : activeSection
            }
            options={visibleSections
              .filter((s) => s !== activeSection)
              .map((label) => ({ label, link: generateColorWallPageUrl(label) }))}
          />
          <button
            className='viewEntire'
            disabled={!visibleSections.includes(primeColorWall) || (primeColorWall === activeSection && !activeFamily)}
            onClick={() => history.push(generateColorWallPageUrl(primeColorWall))}
          >
            <FormattedMessage id='VIEW_ENTIRE_COLOR_WALL' />
          </button>
        </div>
      </div>
    )
  }

  return (
    <AutoSizer disableHeight style={{ width: '100%' }}>
      {({ width }) => (
        <div className={MODE_CLASS_NAMES.BASE}>
          <div className={MODE_CLASS_NAMES.COL}>
            {/* Search and Family Buttons */}
            <div className={MODE_CLASS_NAMES.CELL}>
              <ButtonBar.Bar style={alwaysShowColorFamilies ? { borderRadius: '0' } : {}}>
                {isFamilyView && !alwaysShowColorFamilies ? (
                  <ButtonBar.Button to={generateColorWallPageUrl(section)}>
                    <FontAwesomeIcon className='close-icon-svg' icon={['fa', 'times']} pull='left' />
                    <span className={MODE_CLASS_NAMES.DESC}>
                      <FormattedMessage id='CANCEL' />
                    </span>
                  </ButtonBar.Button>
                ) : (
                  <>
                    {searchColorBtn}
                    {!alwaysShowColorFamilies && families.length > 0 && (
                      <ButtonBar.Button
                        disabled={families.length <= 1}
                        to={`${generateColorWallPageUrl(section)}${PATH_END_FAMILY}`}
                      >
                        <FontAwesomeIcon className='color-families-svg' icon={['fa', 'palette']} pull='left' />
                        <span className={MODE_CLASS_NAMES.DESC}>
                          <FormattedMessage id='COLOR_FAMILIES' />
                        </span>
                      </ButtonBar.Button>
                    )}
                  </>
                )}
              </ButtonBar.Bar>
            </div>
            {/* Prime Color Wall and Collections/Families Categories Buttons */}
            {width > 768 ? (
              <>
                {visibleSections.length > 1 && (!isFamilyView || alwaysShowColorFamilies) && (
                  <div className='menu-bar__border'>
                    {primeColorWall && visibleSections.includes(primeColorWall) && (
                      <NavLink
                        className={`${menuBarPrefix}__prime-color-wall-button ${
                          primeColorWall === activeSection ? 'disabled' : ''
                        }`}
                        to={generateColorWallPageUrl(primeColorWall)}
                      >
                        {primeColorWall}
                      </NavLink>
                    )}
                    {(visibleSections.length > 1 || (isFamilyView && !alwaysShowColorFamilies)) && (
                      <Select
                        activeSection={activeSection}
                        placeholderText={
                          activeSection === primeColorWall
                            ? colorWall.selectSectionText ?? at(messages, 'SELECT_COLLECTION')[0]
                            : activeSection
                        }
                        options={visibleSections
                          .filter((name) => activeFamily !== name && (!primeColorWall || primeColorWall !== name))
                          .map((label) => ({ label, link: generateColorWallPageUrl(label) }))}
                      />
                    )}
                  </div>
                )}
              </>
            ) : (
              // width <= 768
              <div
                className={`menu-bar${
                  visibleSections.length > 1 || isFamilyView || alwaysShowColorFamilies ? '__border' : ''
                }
                ${visibleSections.length > 1 && alwaysShowColorFamilies ? ' menu-bar__border--flex' : ''}`}
              >
                {/* Collections/Families */}
                {(visibleSections.length > 1 || (isFamilyView && !alwaysShowColorFamilies)) && (
                  <Select
                    activeSection={activeSection}
                    placeholderText={
                      isFamilyView && !alwaysShowColorFamilies
                        ? activeFamily ?? at(messages, 'ALL_COLORS')[0]
                        : activeSection
                    }
                    options={((isFamilyView || family) && !alwaysShowColorFamilies ? families : visibleSections)
                      .filter(
                        (name) => activeFamily !== name && (width <= 768 || !primeColorWall || primeColorWall !== name)
                      )
                      .map((label) => ({
                        label,
                        link:
                          isFamilyView && !alwaysShowColorFamilies
                            ? generateColorWallPageUrl(section, label)
                            : generateColorWallPageUrl(label)
                      }))}
                  />
                )}
                {/* Families */}
                {alwaysShowColorFamilies && colorFamilyMenu.current?.clientWidth > width && (
                  <>
                    {visibleSections.length > 1 && <span className='menu-bar__border' />}
                    <Select
                      activeFamily={activeFamily}
                      placeholderText={activeFamily ?? 'All'}
                      disabled={families.length < 1}
                      options={[
                        { label: 'All', link: generateColorWallPageUrl(section) },
                        ...families.map((label) => ({
                          label,
                          link: generateColorWallPageUrl(section, label)
                        }))
                      ]}
                    />
                  </>
                )}
              </div>
            )}
            {shouldShowCloseButton && (!isFamilyView || alwaysShowColorFamilies) && (
              <button
                className='menu-bar__button-close'
                onClick={(e) => {
                  e.preventDefault()
                  dispatch(setIsColorWallModallyPresented())
                  dispatch(navigateToIntendedDestination())
                }}
                style={alwaysShowColorFamilies ? { borderRadius: '0', textTransform: 'uppercase' } : {}}
              >
                {closeBtnShowArrow && (
                  <FontAwesomeIcon className='color-families-svg' icon={['fa', 'times']} pull='left' />
                )}
                <span className={MODE_CLASS_NAMES.DESC}>{closeBtnText ?? <FormattedMessage id='CLOSE' />}</span>
              </button>
            )}
            {/* Color Families Menu Bar Left Side */}
            {isFamilyView && !alwaysShowColorFamilies && width > 768 && (
              <div className={`${MODE_CLASS_NAMES.CELL} ${MODE_CLASS_NAMES.RIGHT}`}>
                <ButtonBar.Bar>
                  <ColorFamilyMenuBtns families={families} section={section} />
                </ButtonBar.Bar>
              </div>
            )}
          </div>
          {/* Color Families Menu Bar Full Size */}
          {alwaysShowColorFamilies && (
            <>
              {/* Color family menu measurer ref - hidden */}
              <div className='color-family-menu color-family-menu__width-size' ref={colorFamilyMenu}>
                <ColorFamilyMenuBtns showAll families={families} section={section} />
              </div>
              {families && colorFamilyMenu.current?.clientWidth < width && (
                <div className='color-family-menu'>
                  <ColorFamilyMenuBtns showAll families={families} section={section} />
                </div>
              )}
            </>
          )}
        </div>
      )}
    </AutoSizer>
  )
}

export default ColorWallToolbar
