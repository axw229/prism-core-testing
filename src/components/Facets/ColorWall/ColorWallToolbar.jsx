/* eslint-disable react/jsx-no-bind */
import React, { useState, useEffect } from 'react'
import { useSelector } from 'react-redux'
import { useRouteMatch, NavLink } from 'react-router-dom'
import { FormattedMessage, useIntl } from 'react-intl'
import { AutoSizer } from 'react-virtualized'
import { Menu, MenuItem, Wrapper, Button } from 'react-aria-menubutton'
import at from 'lodash/at'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { MODE_CLASS_NAMES } from './shared'
import ButtonBar from '../../GeneralButtons/ButtonBar/ButtonBar'
import { generateColorWallPageUrl } from '../../../shared/helpers/ColorUtils'
import './ColorWallMenuBar.scss'

const PATH_END_FAMILY = 'family/'

export default () => {
  const { path, params: { section, family } } = useRouteMatch()
  const { sections = [], families = [] } = useSelector(state => state.colors)
  const activeSection = useSelector(state => state.colors.section)
  const activeFamily = useSelector(state => state.colors.family)
  const isFamilyView = !!family || path.endsWith(PATH_END_FAMILY)
  const menuBarPrefix = 'menu-bar'
  const [menuOpen, setMenuOpen] = useState(false)
  const [showingAllColorFamilies, setShowingAllColorFamilies] = useState(false)
  const { messages = {} } = useIntl()
  const allColorsLabelText = at(messages, 'ALL_COLORS')[0]
  const allCollectionsLabelText = at(messages, 'SELECT_COLLECTION')[0]
  const [currentFamily, setNavMenuButtonText] = useState(allCollectionsLabelText)
  const handleColorFamilySelection = function (value) {
    setNavMenuButtonText(allColorsLabelText)
    setShowingAllColorFamilies(true)
  }

  useEffect(() => {
    if (activeFamily) {
      setShowingAllColorFamilies(false)
      setNavMenuButtonText(activeFamily)
    } else if (activeSection && !showingAllColorFamilies) {
      setNavMenuButtonText(activeSection)
    }
  })

  const colorFamiliesAndSearch = () => {
    return (
      <div className={MODE_CLASS_NAMES.CELL}>
        <ButtonBar.Bar>
          {!isFamilyView && <>
            <ButtonBar.Button disabled={families.length <= 1} onClick={handleColorFamilySelection} to={`${generateColorWallPageUrl(section)}${PATH_END_FAMILY}`}>
              <FontAwesomeIcon className='color-families-svg' icon={['fa', 'palette']} pull='left' />
              <span className={MODE_CLASS_NAMES.DESC}><FormattedMessage id='COLOR_FAMILIES' /></span>
            </ButtonBar.Button>
            <ButtonBar.Button to={`${generateColorWallPageUrl(section, family)}search/`}>
              <FontAwesomeIcon className='color-families-svg' icon={['fa', 'search']} pull='left' />
              <span className={MODE_CLASS_NAMES.DESC}><FormattedMessage id='SEARCH.SEARCH' /></span>
            </ButtonBar.Button>
          </>}
          {!!isFamilyView && <>
            <ButtonBar.Button to={generateColorWallPageUrl(section)}>
              <FontAwesomeIcon className='close-icon-svg' icon={['fa', 'times']} pull='left' />
              <span className={MODE_CLASS_NAMES.DESC}><FormattedMessage id='CANCEL' /></span>
            </ButtonBar.Button>
          </>}
        </ButtonBar.Bar>
      </div>
    )
  }

  return (
    <AutoSizer disableHeight style={{ width: '100%' }}>
      {({ width }) => {
        if (width > 768) {
          return (
            <div className={MODE_CLASS_NAMES.BASE}>
              <div className={MODE_CLASS_NAMES.COL}>
                {colorFamiliesAndSearch()}
                <div className={`${MODE_CLASS_NAMES.CELL} ${MODE_CLASS_NAMES.RIGHT}`}>
                  <ButtonBar.Bar>
                    {(isFamilyView ? families : sections).map(name =>
                      <ButtonBar.Button
                        key={name}
                        to={isFamilyView ? generateColorWallPageUrl(section, name) : generateColorWallPageUrl(name)}
                      >
                        <span className={MODE_CLASS_NAMES.DESC}>{name}</span>
                      </ButtonBar.Button>
                    )}
                  </ButtonBar.Bar>
                </div>
              </div>
            </div>
          )
        } else {
          return (
            <div className={MODE_CLASS_NAMES.BASE}>
              <div className={MODE_CLASS_NAMES.COL}>
                {colorFamiliesAndSearch()}
                <Wrapper
                  className={`${menuBarPrefix} ${menuBarPrefix}--${menuOpen ? 'open' : 'closed'}`}
                  onMenuToggle={({ isOpen }) => { setMenuOpen(isOpen) }}>
                  <Button className={`${menuBarPrefix}__button`} tag='div'>
                    <span className={`${menuBarPrefix}__button-copy`}>{currentFamily}</span>
                    <FontAwesomeIcon className='close-icon-svg' icon={['fa', 'angle-down']} pull='right' />
                  </Button>
                  <Menu className={`${menuBarPrefix}__menu`}>
                    {(isFamilyView || family ? families : sections).map(name =>
                      <MenuItem
                        className={`${menuBarPrefix}__menu-item`}
                        key={name}
                        text={name}
                        value={name}>
                        <NavLink
                          className={`${menuBarPrefix}__menu-link`}
                          to={isFamilyView ? generateColorWallPageUrl(section, name) : generateColorWallPageUrl(name)}
                        >
                          <span className={MODE_CLASS_NAMES.DESC}>{name}</span>
                        </NavLink>
                      </MenuItem>
                    )}
                  </Menu>
                </Wrapper>
              </div>
            </div>
          )
        }
      }}
    </AutoSizer>
  )
}
