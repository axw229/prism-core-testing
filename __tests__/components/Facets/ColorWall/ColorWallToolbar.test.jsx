import React from 'react'
import ColorWallToolbar from 'src/components/Facets/ColorWall/ColorWallToolbar'

test('snapshot matches', () => {
  expect(mocked(<ColorWallToolbar />)).toMatchSnapshot()
})
