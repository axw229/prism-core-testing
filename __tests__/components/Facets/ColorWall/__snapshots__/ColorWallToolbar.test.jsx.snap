// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot matches 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  locale="en-US"
  messages={
    Object {
      "ADD_A_COLOR": "ADD A COLOR",
      "BACK_TO_COLOR_WALL": "Back to Color Wall",
      "CANCEL": "Cancel",
      "CANVAS_UNSUPPORTED": "Your browser does not support the canvas element.",
      "CHIP_MAXIMIZED": "Color Chip has been maximized",
      "CHIP_MINIMIZED": "Color Chip has been minimized",
      "COLOR_COLLECTIONS": "Color Collections",
      "COLOR_FAMILIES": "Color Families",
      "COORDINATING_COLORS": "Coordinating Colors",
      "DELETE_PIN": "Delete pin",
      "DETAILS": "Details",
      "DISPLAY_SCENE_PAINTER": "Display the Scene Painter",
      "ERROR_LOADING_COLORS": "We're unable to load colors right now.",
      "EXTERIOR": "Exterior",
      "FIND_COLORS_IN_CW": "FIND COLORS IN THE DIGITAL COLOR WALL",
      "HIDE_SCENE_PAINTER": "Hide the Scene Painter",
      "IMAGE_INVISIBLE": "This image is invisible to the user.",
      "INTERIOR": "Interior",
      "LOCATION_NUMBER": "Location Number",
      "MAXIMIZE_COLOR_SWATCH": "Maximize the Color Swatch",
      "NO_COLORS_AVAILABLE": "No colors available.",
      "PAINT_TOOLS.BACK": "Go back",
      "PAINT_TOOLS.CLOSE": "Close tooltip",
      "PAINT_TOOLS.DEFINEAREA": "Create a polygon to fill",
      "PAINT_TOOLS.DELETEGROUP": "Delete the group",
      "PAINT_TOOLS.ERASE": "Remove paint by erasing",
      "PAINT_TOOLS.FORWARD": "Go forward",
      "PAINT_TOOLS.GROUP": "Group the selections",
      "PAINT_TOOLS.HIDEPAINT": "Hide the painted area",
      "PAINT_TOOLS.INFO": "Toggle the tool tips",
      "PAINT_TOOLS.PAINTAREA": "fill a continuous area with color",
      "PAINT_TOOLS.PAINTBRUSH": "Paint using a brush",
      "PAINT_TOOLS.REDO": "Redo an action",
      "PAINT_TOOLS.REMOVEAREA": "Erase an area by creating a polygon",
      "PAINT_TOOLS.SELECTAREA": "Select an area to manipulate",
      "PAINT_TOOLS.TOOLTIPS.DEFINEAREA": "Outline an area to paint it. Be sure to enclose the area by starting and ending at the same point.",
      "PAINT_TOOLS.TOOLTIPS.ERASE": "Remove paint from your scene.",
      "PAINT_TOOLS.TOOLTIPS.HIDEPAINT": "Use this tool to see the original room scene.",
      "PAINT_TOOLS.TOOLTIPS.INFO": "Click here to revisit the Tool Tips.",
      "PAINT_TOOLS.TOOLTIPS.PAINTAREA": "Use the Paint Area feature to automatically detect and paint surface areas. Just click or tap on a surface to highlight it, then click or tap again.",
      "PAINT_TOOLS.TOOLTIPS.PAINTBRUSH": "Allows for the freehand painting of your selected scene. Just point, click or tap and drag to add color to any area.",
      "PAINT_TOOLS.TOOLTIPS.REDO": "Step forward from your last action.",
      "PAINT_TOOLS.TOOLTIPS.REMOVEAREA": "Use this tool to select and remove paint from a defined area.",
      "PAINT_TOOLS.TOOLTIPS.SELECTAREA": "Select an area, then use the tool bar to paint, edit paint colors or remove paint from your scene.",
      "PAINT_TOOLS.TOOLTIPS.SELECTGROUP": "Use this tool to group and edit multiple areas at the same time. Use the Delete tool to remove areas from your scene.",
      "PAINT_TOOLS.TOOLTIPS.UNDO": "Step backwards from your last action.",
      "PAINT_TOOLS.TOOLTIPS.ZOOM": "Zoom in and pan around within your scene to paint small areas with increased accuracy.",
      "PAINT_TOOLS.UNDO": "Undo an action",
      "PAINT_TOOLS.UNGROUP": "Ungroup the selections",
      "PAINT_TOOLS.ZOOM": "Adjust the zoom level",
      "PALETTE_TITLE": "My Color Palette",
      "RESTORE_COLOR_SWATCH_TO_DEFAULT_SIZE": "Restore the Color Swatch to default size",
      "SCENE_DISPLAYED": "Scene is displayed",
      "SCENE_HIDDEN": "Scene is hidden",
      "SEARCH.ERROR.GENERIC": "We were unable to perform your search.",
      "SEARCH.ERROR.HEADLINE": "Something went wrong.",
      "SEARCH.FIND_A_COLOR": "Find a color",
      "SEARCH.NO_RESULTS": "Sorry, no color matches found.",
      "SEARCH.PROMPT": "Enter a color name, number, or family in the text field above.",
      "SEARCH.SEARCH": "Search",
      "SEARCH.SEARCHING": "Searching",
      "SEARCH.SEARCH_BY": "Search by color name or number",
      "SEARCH.SUGGESTIONS": "Some examples of things you can search for are {suggestions}",
      "SIMILAR_COLORS": "Similar Colors",
      "SWATCH_ACTIVATED_DETAILS": "{color} is selected. Press Enter to view details.",
      "SWATCH_FOCUS": "{color}. Press Enter to select.",
      "TO_DAY_VIEW": "Switch to day view",
      "TO_NIGHT_VIEW": "Switch to night view",
      "VIEW_DETAILS": "View Details",
      "ZOOM_IN": "Zoom in",
      "ZOOM_OUT": "Zoom out",
    }
  }
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <_default>
        <div
          className="color-wall-mode-btns"
        >
          <div
            className="color-wall-mode-btns__col"
          >
            <div
              className="color-wall-mode-btns__cell"
            >
              <Bar>
                <div
                  className="button-bar__option-container auto-scroll"
                >
                  <ul
                    className="button-bar__options no-list"
                  />
                </div>
              </Bar>
            </div>
            <div
              className="color-wall-mode-btns__cell color-wall-mode-btns__right"
            >
              <Bar>
                <div
                  className="button-bar__option-container auto-scroll"
                >
                  <ul
                    className="button-bar__options no-list"
                  >
                    <Button
                      disabled={true}
                      to="/active/color-wall/family/"
                    >
                      <li
                        className="button-bar__options__option"
                      >
                        <NavLink
                          activeClassName="button-bar__options__option__btn--active "
                          className="button-bar__options__option__btn "
                          disabled={true}
                          onClick={null}
                          to="/active/color-wall/family/"
                        >
                          <Link
                            aria-current={null}
                            className="button-bar__options__option__btn "
                            disabled={true}
                            onClick={null}
                            to={
                              Object {
                                "hash": "",
                                "pathname": "/active/color-wall/family/",
                                "search": "",
                                "state": null,
                              }
                            }
                          >
                            <LinkAnchor
                              aria-current={null}
                              className="button-bar__options__option__btn "
                              disabled={true}
                              href="/active/color-wall/family/"
                              navigate={[Function]}
                              onClick={null}
                            >
                              <a
                                aria-current={null}
                                className="button-bar__options__option__btn "
                                disabled={true}
                                href="/active/color-wall/family/"
                                onClick={[Function]}
                              >
                                <FontAwesomeIcon
                                  border={false}
                                  className="color-families-svg"
                                  fixedWidth={false}
                                  flip={null}
                                  icon={
                                    Array [
                                      "fa",
                                      "palette",
                                    ]
                                  }
                                  inverse={false}
                                  listItem={false}
                                  mask={null}
                                  pull="left"
                                  pulse={false}
                                  rotation={null}
                                  size={null}
                                  spin={false}
                                  swapOpacity={false}
                                  symbol={false}
                                  title=""
                                  transform={null}
                                >
                                  <svg
                                    aria-hidden="true"
                                    className="svg-inline--fa fa-palette fa-w-16 fa-pull-left color-families-svg"
                                    data-icon="palette"
                                    data-prefix="fa"
                                    focusable="false"
                                    role="img"
                                    style={Object {}}
                                    viewBox="0 0 512 512"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M204.3 5C104.9 24.4 24.8 104.3 5.2 203.4c-37 187 131.7 326.4 258.8 306.7 41.2-6.4 61.4-54.6 42.5-91.7-23.1-45.4 9.9-98.4 60.9-98.4h79.7c35.8 0 64.8-29.6 64.9-65.3C511.5 97.1 368.1-26.9 204.3 5zM96 320c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm32-128c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128-64c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128 64c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32z"
                                      fill="currentColor"
                                      style={Object {}}
                                    />
                                  </svg>
                                </FontAwesomeIcon>
                                <span
                                  className="color-wall-mode-btns__descendant inherit-font"
                                >
                                  <FormattedMessage
                                    id="COLOR_FAMILIES"
                                    values={Object {}}
                                  >
                                    Color Families
                                  </FormattedMessage>
                                </span>
                              </a>
                            </LinkAnchor>
                          </Link>
                        </NavLink>
                      </li>
                    </Button>
                    <Button
                      to="/active/color-wall/search/"
                    >
                      <li
                        className="button-bar__options__option"
                      >
                        <NavLink
                          activeClassName="button-bar__options__option__btn--active "
                          className="button-bar__options__option__btn "
                          onClick={null}
                          to="/active/color-wall/search/"
                        >
                          <Link
                            aria-current={null}
                            className="button-bar__options__option__btn "
                            onClick={null}
                            to={
                              Object {
                                "hash": "",
                                "pathname": "/active/color-wall/search/",
                                "search": "",
                                "state": null,
                              }
                            }
                          >
                            <LinkAnchor
                              aria-current={null}
                              className="button-bar__options__option__btn "
                              href="/active/color-wall/search/"
                              navigate={[Function]}
                              onClick={null}
                            >
                              <a
                                aria-current={null}
                                className="button-bar__options__option__btn "
                                href="/active/color-wall/search/"
                                onClick={[Function]}
                              >
                                <FontAwesomeIcon
                                  border={false}
                                  className="color-families-svg"
                                  fixedWidth={false}
                                  flip={null}
                                  icon={
                                    Array [
                                      "fa",
                                      "search",
                                    ]
                                  }
                                  inverse={false}
                                  listItem={false}
                                  mask={null}
                                  pull="left"
                                  pulse={false}
                                  rotation={null}
                                  size={null}
                                  spin={false}
                                  swapOpacity={false}
                                  symbol={false}
                                  title=""
                                  transform={null}
                                >
                                  <svg
                                    aria-hidden="true"
                                    className="svg-inline--fa fa-search fa-w-16 fa-pull-left color-families-svg"
                                    data-icon="search"
                                    data-prefix="fa"
                                    focusable="false"
                                    role="img"
                                    style={Object {}}
                                    viewBox="0 0 512 512"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                                      fill="currentColor"
                                      style={Object {}}
                                    />
                                  </svg>
                                </FontAwesomeIcon>
                                <span
                                  className="color-wall-mode-btns__descendant inherit-font"
                                >
                                  <FormattedMessage
                                    id="SEARCH.SEARCH"
                                    values={Object {}}
                                  >
                                    Search
                                  </FormattedMessage>
                                </span>
                              </a>
                            </LinkAnchor>
                          </Link>
                        </NavLink>
                      </li>
                    </Button>
                  </ul>
                </div>
              </Bar>
            </div>
          </div>
        </div>
      </_default>
    </Router>
  </Provider>
</IntlProvider>
`;
